app/models/ordergroup.rb

```diff
@@ -39,6 +39,12 @@ def self.include_transaction_class_sum
       .group('groups.id')
   end
 
+  def self.custom_fields
+    fields = FoodsoftConfig[:custom_fields] && FoodsoftConfig[:custom_fields][:ordergroup]
```

Comment:
it seems like this would be a common check required - why not move the logic inside `FoodsoftConfig`

part of me wonders why not go with `FoodsoftConfig[:ordergroup]` and have it looked up by missing_method.  but this is a late comment.  

i would still recommend a specific getter for custom fields so you don't have to do the nil check.  (or alternatively, populate `FoodsoftConfig[:custom_fields]={}` so you can avoid these checks and could have the more terse `return [] unless FoodsoftConfig[:custom_fields][:ordergroup]`

Reply:
indeed, if you are going with the `ApplicationRecord` pattern, then `self.custom_fields` seems like it would be better put in there (adding custom_fields support to all models)

Reply:
I agree that `FoodsoftSettings` could be refactored to better support nested settings. This would include, I guess. And then we could convert these hashes to properly nested settings as well. That's some work, so I'm ok not doing that here.

On another note, what about expanding the `CustomFields` to something like (not tested, but you'll get the gist):
```ruby
module CustomFields
  extend ActiveSupport::Concern
  include RailsSettings::Extend

  class_methods do
    def has_custom_fields(key)
      attr_accessor :custom_fields

      after_initialize do
        settings.defaults['custom_fields'] = { } unless settings.custom_fields
      end

      after_save do
        self.settings.custom_fields = custom_fields if custom_fields
      end

      def custom_fields
        @custom_fields ||= begin
          fields = FoodsoftConfig[:custom_fields] && FoodsoftConfig[:custom_fields][key]
          fields ||= []
          fields.map(&:deep_symbolize_keys)
        end
      end
    end
  end
end
```